src/imageops/sample.rs

```diff
@@ -750,16 +750,28 @@ where
 /// Resize the supplied image to the specified dimensions.
 /// ```nwidth``` and ```nheight``` are the new dimensions.
 /// ```filter``` is the sampling filter to use.
-pub fn resize<I: GenericImageView>(
+pub fn resize<I, P, S>(
     image: &I,
     nwidth: u32,
     nheight: u32,
     filter: FilterType,
-) -> ImageBuffer<I::Pixel, Vec<<I::Pixel as Pixel>::Subpixel>>
+) -> ImageBuffer<P, Vec<S>>
 where
-    I::Pixel: 'static,
-    <I::Pixel as Pixel>::Subpixel: 'static,
+    I: GenericImageView<Pixel = P>,
+    P: Pixel<Subpixel = S> + 'static,
+    S: Primitive + 'static,
 {
+    // check if the new dimensions are the same as the old. if they are, make a copy instead of resampling
+    if (nwidth, nheight) == image.dimensions() {
+        let mut tmp = ImageBuffer::<P, Vec<S>>::new(image.width(), image.height());
```

Comment:
I think the compiler should infer the type here so you should be able to do `ImageBuffer::new(...)`